### PR TITLE
Add cloudbuild configuration.

### PR DESCRIPTION
**Pull request by nanliu**

### CODE REVIEW

**Comment by nanliu** [comment]
This provides a config for building and deploying this as a container into gcr.io.

**Comment by luna-duclos** [comment]
Could you add some instructions on how to use this ?

**Comment by luna-duclos** [comment]
See https://github.com/PalmStoneGames/kube-cert-manager/blob/master/codebuild/README.md for an example on how AWS cloudbuild is documented.
We also probably want to make the distinction clear between google and AWS cloudbuild somehow

**Comment by nanliu** [comment]
I've included some documentation. Let me know if it's sufficient.

**Comment by luna-duclos** [comment]
Will merge this soon, probably tomorrow.

File: CONTAINER_BUILDER.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+* select [Container Registry](https://cloud.google.com/container-registry/) > [build triggers](https://cloud.google.com/container-builder/docs/creating-build-triggers)
+* select GitHub as source and authenticate
+* select your fork of kube-cert-manager repostory
+* create trigger and use cloudbuild.yaml for build configuration
+
+![container_builder](./docs/container_builder.png)
+
+When changes are pushed to your repository, Container Builder will build and publish a new container to `gcr.io/${PROJECT_ID}/kube-cert-manager`. The image will be tagged with git sha1 and `latest` for convenience.
+
```
Comment (ahmetb) [inline CONTAINER_BUILDER.md:17]: Can you also add manual one-off build command please? This doesn't require forking the repo.

    gcloud container builds submit --config cloudbuild.yaml .

**Review by ahmetb — CHANGES_REQUESTED** [review]

File: CONTAINER_BUILDER.md (hunk 1 above)
Comment (nanliu) [reply CONTAINER_BUILDER.md:17]: This has been added to the section below.

File: CONTAINER_BUILDER.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+## Container Builder CLI
+
+To kick off a container builder job with your local copy of the source code:
+```
+$ git clone https://github.com/PalmStoneGames/kube-cert-manager.git
+$ cd kube-cert-manager
+$ gcloud container builds submit --config cloudbuild.yaml .
+```
+
+To review jobs created by container builder:
```
Comment (ahmetb) [inline CONTAINER_BUILDER.md:27]: @nanliu you don't need this as `gcloud container builds submit ` streams the build log and the result.

File: CONTAINER_BUILDER.md (hunk 1 above)
Comment (ahmetb) [reply CONTAINER_BUILDER.md:17]: thanks

File: CONTAINER_BUILDER.md (hunk 2 above)
Comment (nanliu) [reply CONTAINER_BUILDER.md:27]: This is for jobs triggered by github PR and not submitted manually.

Comment (ahmetb) [reply CONTAINER_BUILDER.md:27]: You're listing it under the CLI build though. Which gives the impression that manual build submit and listing the build result are related somewhat :)

**Pull request merged**